**NovaFit Google Ads Campaign Performance Analysis**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df = pd.read_csv('/content/NovaFit_Search_Strategy_Toolkit - Search_Performance_Data.csv')

df.head()

,Date,Campaign_Type,Campaign,Ad_Group,Keyword,Match_Type,Impressions,Clicks,Cost_COP,Conversions,Quality_Score
0,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit,Exact,435.0,44,$39.115,4,9
1,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit colombia,Phrase,215.0,20,$21.774,2,10
2,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit tienda oficial,Phrase,77.0,11,$9.851,2,10
3,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit medellin,Phrase,87.0,10,$9.331,1,8
4,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,tienda novafit,Phrase,59.0,5,$5.992,0,8


In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 720 entries, 0 to 719
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Date           720 non-null    object 
 1   Campaign_Type  720 non-null    object 
 2   Campaign       720 non-null    object 
 3   Ad_Group       720 non-null    object 
 4   Keyword        720 non-null    object 
 5   Match_Type     720 non-null    object 
 6   Impressions    720 non-null    float64
 7   Clicks         720 non-null    int64  
 8   Cost_COP       720 non-null    object 
 9   Conversions    720 non-null    int64  
 10  Quality_Score  720 non-null    int64  
dtypes: float64(1), int64(3), object(7)
memory usage: 62.0+ KB


In [4]:
df.shape

(720, 11)

In [5]:
df.describe()

,Impressions,Clicks,Conversions,Quality_Score
count,720.000000,720.000000,720.000000,720.000000
mean,289.910544,16.413889,0.906944,6.422222
std,267.493926,16.835317,1.624211,2.094979
min,1.005000,0.000000,0.000000,3.000000
25%,81.000000,3.000000,0.000000,5.000000
50%,182.000000,12.000000,0.000000,6.000000
75%,462.000000,22.000000,1.000000,8.000000
max,994.000000,106.000000,12.000000,10.000000


In [6]:
df.isnull().sum()

,0
Date,0
Campaign_Type,0
Campaign,0
Ad_Group,0
Keyword,0
Match_Type,0
Impressions,0
Clicks,0
Cost_COP,0
Conversions,0


In [7]:
df.duplicated().sum()

np.int64(0)

In [9]:
df.columns.tolist()

['Date',
 'Campaign_Type',
 'Campaign',
 'Ad_Group',
 'Keyword',
 'Match_Type',
 'Impressions',
 'Clicks',
 'Cost_COP',
 'Conversions',
 'Quality_Score']

In [10]:
print("Campaign Types:")
print(df['Campaign_Type'].value_counts())

print("\nMatch Types:")
print(df['Match_Type'].value_counts())

print("\nCampaigns:")
print(df['Campaign'].value_counts())

print("\nAd Groups:")
print(df['Ad_Group'].value_counts())

Campaign Types:
Campaign_Type
Category      360
Brand         180
Competitor    180
Name: count, dtype: int64

Match Types:
Match_Type
Phrase    480
Broad     210
Exact      30
Name: count, dtype: int64

Campaigns:
Campaign
NovaFit - Category      360
NovaFit - Brand         180
NovaFit - Competitor    180
Name: count, dtype: int64

Ad Groups:
Ad_Group
NovaFit Brand Terms    180
Leggings                90
Training Shoes          90
Gym Membership          90
Supplements             90
FitZone                 90
UrbanAthletics          90
Name: count, dtype: int64


In [11]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 720 entries, 0 to 719
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Date           720 non-null    object 
 1   Campaign_Type  720 non-null    object 
 2   Campaign       720 non-null    object 
 3   Ad_Group       720 non-null    object 
 4   Keyword        720 non-null    object 
 5   Match_Type     720 non-null    object 
 6   Impressions    720 non-null    float64
 7   Clicks         720 non-null    int64  
 8   Cost_COP       720 non-null    object 
 9   Conversions    720 non-null    int64  
 10  Quality_Score  720 non-null    int64  
dtypes: float64(1), int64(3), object(7)
memory usage: 62.0+ KB


In [12]:
print(df.columns.tolist())

['Date', 'Campaign_Type', 'Campaign', 'Ad_Group', 'Keyword', 'Match_Type', 'Impressions', 'Clicks', 'Cost_COP', 'Conversions', 'Quality_Score']


In [13]:
df.isnull().sum()

,0
Date,0
Campaign_Type,0
Campaign,0
Ad_Group,0
Keyword,0
Match_Type,0
Impressions,0
Clicks,0
Cost_COP,0
Conversions,0


In [14]:
df.duplicated().sum()

np.int64(0)

In [18]:
df_clean = df.copy()

In [17]:
df_clean['Date'] = pd.to_datetime(df_clean['Date'])

In [19]:
df_clean['Date'] = pd.to_datetime(df_clean['Date'])

In [20]:
df_clean['Date'].dtype

dtype('<M8[ns]')

In [22]:
df.columns.tolist()

['Date',
 'Campaign_Type',
 'Campaign',
 'Ad_Group',
 'Keyword',
 'Match_Type',
 'Impressions',
 'Clicks',
 'Cost_COP',
 'Conversions',
 'Quality_Score']

In [23]:
numeric_columns = [
    'Impressions',
    'Clicks',
    'Cost_COP',
    'Conversions',
    'Quality_Score'
]

df_clean[numeric_columns] = df_clean[numeric_columns].apply(
    pd.to_numeric,
    errors='coerce'
)

In [24]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 720 entries, 0 to 719
Data columns (total 11 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Date           720 non-null    datetime64[ns]
 1   Campaign_Type  720 non-null    object        
 2   Campaign       720 non-null    object        
 3   Ad_Group       720 non-null    object        
 4   Keyword        720 non-null    object        
 5   Match_Type     720 non-null    object        
 6   Impressions    720 non-null    float64       
 7   Clicks         720 non-null    int64         
 8   Cost_COP       0 non-null      float64       
 9   Conversions    720 non-null    int64         
 10  Quality_Score  720 non-null    int64         
dtypes: datetime64[ns](1), float64(2), int64(3), object(5)
memory usage: 62.0+ KB


In [25]:
df_clean.isnull().sum()

,0
Date,0
Campaign_Type,0
Campaign,0
Ad_Group,0
Keyword,0
Match_Type,0
Impressions,0
Clicks,0
Cost_COP,720
Conversions,0


In [26]:
print(df['Cost_COP'].head(10))
print(df['Cost_COP'].dtype)

0    $39.115
1    $21.774
2     $9.851
3     $9.331
4     $5.992
5     $3.291
6    $72.881
7    $48.964
8    $19.617
9    $40.574
Name: Cost_COP, dtype: object
object


In [27]:
df['Cost_COP'].head(10)

,Cost_COP
0,$39.115
1,$21.774
2,$9.851
3,$9.331
4,$5.992
5,$3.291
6,$72.881
7,$48.964
8,$19.617
9,$40.574


In [30]:
df_clean = df.copy()

In [31]:
df_clean['Date'] = pd.to_datetime(df_clean['Date'])

In [32]:
df_clean['Cost_COP'] = (
    df['Cost_COP']
    .str.replace('$', '', regex=False)
    .str.replace('.', '', regex=False)
    .astype(float)
)

In [33]:
df_clean['Cost_COP'].head(10)

,Cost_COP
0,39115.0
1,21774.0
2,9851.0
3,9331.0
4,5992.0
5,3291.0
6,72881.0
7,48964.0
8,19617.0
9,40574.0


In [34]:
df_clean['Cost_COP'].dtype

dtype('float64')

In [35]:
df_clean['Cost_COP'].isnull().sum()

np.int64(0)

In [36]:
numeric_columns = [
    'Impressions',
    'Clicks',
    'Conversions',
    'Quality_Score'
]

df_clean[numeric_columns] = df_clean[numeric_columns].apply(
    pd.to_numeric,
    errors='coerce'
)

In [37]:
df_clean.isnull().sum()

,0
Date,0
Campaign_Type,0
Campaign,0
Ad_Group,0
Keyword,0
Match_Type,0
Impressions,0
Clicks,0
Cost_COP,0
Conversions,0


In [38]:
df_clean['CTR'] = np.where(
    df_clean['Impressions'] > 0,
    (df_clean['Clicks'] / df_clean['Impressions']) * 100,
    0
)

df_clean['CPC'] = np.where(
    df_clean['Clicks'] > 0,
    df_clean['Cost_COP'] / df_clean['Clicks'],
    0
)

df_clean['CPA'] = np.where(
    df_clean['Conversions'] > 0,
    df_clean['Cost_COP'] / df_clean['Conversions'],
    0
)

df_clean['Conversion_Rate'] = np.where(
    df_clean['Clicks'] > 0,
    (df_clean['Conversions'] / df_clean['Clicks']) * 100,
    0
)

In [39]:
df_clean.head()

,Date,Campaign_Type,Campaign,Ad_Group,Keyword,Match_Type,Impressions,Clicks,Cost_COP,Conversions,Quality_Score,CTR,CPC,CPA,Conversion_Rate
0,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit,Exact,435.0,44,39115.0,4,9,10.114943,888.977273,9778.75,9.090909
1,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit colombia,Phrase,215.0,20,21774.0,2,10,9.302326,1088.700000,10887.00,10.000000
2,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit tienda oficial,Phrase,77.0,11,9851.0,2,10,14.285714,895.545455,4925.50,18.181818
3,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,novafit medellin,Phrase,87.0,10,9331.0,1,8,11.494253,933.100000,9331.00,10.000000
4,2026-07-12,Brand,NovaFit - Brand,NovaFit Brand Terms,tienda novafit,Phrase,59.0,5,5992.0,0,8,8.474576,1198.400000,0.00,0.000000


In [40]:
df_clean[['CTR', 'CPC', 'CPA', 'Conversion_Rate']].describe()

,CTR,CPC,CPA,Conversion_Rate
count,720.000000,720.000000,720.000000,720.000000
mean,383.279442,1968.543096,14093.362021,3.510561
std,966.732486,805.440777,21124.022313,4.607521
min,0.000000,0.000000,0.000000,0.000000
25%,1.639344,1202.250000,0.000000,0.000000
50%,3.174603,1896.089069,0.000000,0.000000
75%,9.772880,2331.003788,25251.375000,5.882353
max,4784.688995,3731.000000,104936.000000,20.000000


In [41]:
campaign_type_summary = df_clean.groupby('Campaign_Type').agg({
    'Impressions': 'sum',
    'Clicks': 'sum',
    'Cost_COP': 'sum',
    'Conversions': 'sum'
}).reset_index()

campaign_type_summary

,Campaign_Type,Impressions,Clicks,Cost_COP,Conversions
0,Brand,36530.000,3858,3700915.0,396
1,Category,144730.592,7646,14374023.0,257
2,Competitor,27475.000,314,992595.0,0


In [42]:
campaign_type_summary['CTR'] = (
    campaign_type_summary['Clicks'] /
    campaign_type_summary['Impressions']
) * 100

campaign_type_summary['CPC'] = (
    campaign_type_summary['Cost_COP'] /
    campaign_type_summary['Clicks']
)

campaign_type_summary['CPA'] = (
    campaign_type_summary['Cost_COP'] /
    campaign_type_summary['Conversions']
)

campaign_type_summary['Conversion_Rate'] = (
    campaign_type_summary['Conversions'] /
    campaign_type_summary['Clicks']
) * 100

campaign_type_summary

,Campaign_Type,Impressions,Clicks,Cost_COP,Conversions,CTR,CPC,CPA,Conversion_Rate
0,Brand,36530.000,3858,3700915.0,396,10.561183,959.283307,9.345745e+03,10.264386
1,Category,144730.592,7646,14374023.0,257,5.282919,1879.940230,5.593005e+04,3.361235
2,Competitor,27475.000,314,992595.0,0,1.142857,3161.130573,inf,0.000000


In [43]:
campaign_type_summary.round(2)

,Campaign_Type,Impressions,Clicks,Cost_COP,Conversions,CTR,CPC,CPA,Conversion_Rate
0,Brand,36530.00,3858,3700915.0,396,10.56,959.28,9345.74,10.26
1,Category,144730.59,7646,14374023.0,257,5.28,1879.94,55930.05,3.36
2,Competitor,27475.00,314,992595.0,0,1.14,3161.13,inf,0.00


**Campaign Performance Analysis**

The analysis shows clear differences in performance across NovaFit's three campaign types: Brand, Category and Competitor.

### Brand Campaign

The Brand campaign was the strongest-performing campaign overall. It generated 396 conversions from 36,530 impressions and 3,858 clicks, while spending 3,700,915 COP. It achieved the highest CTR (10.56%), lowest CPC (959.28 COP) and lowest CPA (9,345.74 COP) of the three campaigns. Its 10.26% conversion rate was also substantially higher than Category and Competitor.

These results suggest that users searching for NovaFit-related terms were highly engaged and more likely to convert, making Brand the most efficient campaign in the dataset.

### Category Campaign

The Category campaign generated the highest traffic volume, receiving 144,730.59 impressions and 7,646 clicks. It also accounted for the largest amount of spend at 14,374,023 COP and generated 257 conversions.

However, its efficiency was considerably lower than Brand. Category achieved a 5.28% CTR, 1,879.94 COP CPC, 55,930.05 COP CPA and 3.36% conversion rate. Although it generated a significant number of conversions, the much higher CPA indicates that these conversions were acquired at a substantially greater cost than Brand conversions.

This suggests that Category provides valuable acquisition potential but may require optimisation to improve conversion efficiency.

### Competitor Campaign

The Competitor campaign performed the weakest overall. It generated 27,475 impressions and 314 clicks while spending 992,595 COP, but produced zero conversions. Its 1.14% CTR was the lowest of all three campaigns, while its 3,161.13 COP CPC was the highest.

As there were no conversions, a CPA cannot be calculated meaningfully. The 0% conversion rate further indicates that the traffic generated by the Competitor campaign did not result in conversions during the period analysed.

### Overall Findings

Overall, Brand was the most efficient campaign, combining the highest conversion volume with the lowest CPC and CPA and the highest CTR and conversion rate. Category generated the greatest amount of traffic and spend, but its substantially higher CPA indicates that there is room to improve efficiency. Competitor was the weakest campaign, generating clicks but no conversions.

From a strategic perspective, the results suggest that NovaFit should protect the strong performance of Brand, optimise Category to improve its conversion efficiency, and carefully review the continued allocation of spend to Competitor given its lack of conversions.

## Key Findings

- **Brand was the most efficient campaign**, achieving the highest CTR (10.56%), highest conversion rate (10.26%), and lowest CPA (9,345.74 COP).
- **Category generated the greatest volume**, with 144,730.59 impressions, 7,646 clicks and 257 conversions, but had a substantially higher CPA of 55,930.05 COP.
- **Competitor generated no conversions** despite spending 992,595 COP, with the lowest CTR (1.14%) and highest CPC (3,161.13 COP).
- The results indicate that Brand currently provides the strongest return from search activity, while Category represents an opportunity for optimisation and Competitor requires careful budget consideration.

## Recommendations

### 1. Protect Brand performance

Brand should continue to receive sufficient budget to capture high-intent searches because it generated the highest number of conversions at the lowest CPA.

### 2. Optimise Category performance

Category generates substantial traffic and conversions but at a much higher CPA than Brand. Keyword, ad and landing-page performance should therefore be reviewed to identify opportunities to improve conversion efficiency.

### 3. Review Competitor spend

The Competitor campaign generated 314 clicks but no conversions. Given its 0% conversion rate and highest CPC, its budget should be carefully reviewed before increasing spend.

### 4. Monitor campaign efficiency

Future optimisation should consider CPA, conversion rate, CTR and CPC together rather than relying on conversion volume alone. This would help NovaFit balance customer acquisition volume with cost efficiency.